### PR TITLE
Update to nix 2.3.4

### PR DESCRIPTION
**Pull request by prusnak**
This change also includes an addition of ARG instruction, so the installed nix version could be easily changed via `docker build --build-arg NIX_VERSION=<version>`

Supersedes https://github.com/NixOS/docker/pull/14

### CODE REVIEW

**Comment by domenkozar** [comment]
@peti I'm not sure how you test Nix upgrades, but a simple `nix-build -A git '<nixpkgs>'` works with this commit and it would be good to merge this.

File: Dockerfile [hunk 1]
```diff
@@ -3,19 +3,22 @@
 FROM alpine
 
 # Enable HTTPS support in wget and set nsswitch.conf to make resolution work within containers
-RUN apk add --no-cache --update openssl \
+RUN : \
```
Comment (peti) [inline Dockerfile:6]: Please don't submit unrelated formatting changes in a version update. I don't like this style very much, to be honest, and this now prevents me from merging the actual update, which I do like.

**Comment by peti** [comment]
> I'm not sure how you test Nix upgrades.

Travis does run a test (and it's succeeding).

File: Dockerfile (hunk 1 above)
Comment (prusnak) [reply Dockerfile]: I thought you might like it. Amended in adb72e8a1e40ab78b3551f8d0ac3f6d423bd7e97

**Comment by prusnak** [comment]
Thanks peti for the merge and for pushing the updated container to Docker Hub!

Can we maybe tag the container with `2.3.4` on Docker Hub?

**Comment by domenkozar** [comment]
I've just done that, should be available in a few minutes.

**Comment by peti** [comment]
The tag now exists: https://hub.docker.com/repository/docker/nixos/nix
